### PR TITLE
frontend/feat/#1063: added clevertap properties

### PR DESCRIPTION
**Pull request by ashkriti**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Integrated API for the properties required for CleverTap
Added the push templets for CleverTap notification

## Motivation and Context
https://github.com/nammayatri/nammayatri/issues/1063

## How did you test it?
Tested Manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Flow.purs [hunk 1]
```diff
@@ -713,6 +740,10 @@ homeScreenFlow = do
         modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = SearchLocationModel ,rideRequestFlow = false, isSearchLocation = SearchLocation, isSrcServiceable = false, isSource = Just true, isRideServiceable = false}})
         homeScreenFlow
         else pure unit
+      let currentTime = convertTo12HourFormat (convertUTCtoISC (getCurrentUTC "") "HH:mm:ss")
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:743]: try to use `convertTo12HourFormat` existing function

File: Frontend/ui-customer/src/Flow.purs [hunk 2]
```diff
@@ -424,7 +424,34 @@ currentFlowStatus = do
       setValueToLocalStore DISABILITY_UPDATED $ if (isNothing response.hasDisability) then "false" else "true"
       setValueToLocalStore REFERRAL_STATUS  $ if response.hasTakenRide then "HAS_TAKEN_RIDE" else if (response.referralCode /= Nothing && not response.hasTakenRide) then "REFERRED_NOT_TAKEN_RIDE" else "NOT_REFERRED_NOT_TAKEN_RIDE"
       setValueToLocalStore HAS_TAKEN_FIRST_RIDE if response.hasTakenRide then "true" else "false"
-      void $ pure $ setCleverTapUserProp "First ride taken" if response.hasTakenRide then "true" else "false"
+      (PersonStatsRes resp) <- Remote.getPersonStatsBT ""
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:427]: move it to a separate function

File: Frontend/ui-customer/src/Screens/CustomerUtils/InvoiceScreen/Controller.purs [hunk 3]
```diff
@@ -69,11 +70,11 @@ eval (GenericHeaderAC (GenericHeaderController.PrefixImgOnClick)) state = contin
...
   continueWithCmd state
     [ do
-        let _ = unsafePerformEffect $ logEventWithMultipleParams state.data.logField "ny_user_invoice_download" $ [ { key : "Base fare", value : state.data.selectedItem.baseFare},
-                                                                                                                    { key : "Distance", value : state.data.selectedItem.baseDistance},
-                                                                                                                    { key : "Driver pickup charges", value : "₹ 10"},
-                                                                                                                    { key : "Total fare", value : state.data.selectedItem.totalAmount},
-                                                                                                                    { key : "Ride completion timestamp", value : state.data.selectedItem.rideEndTime}]
+        let _ = unsafePerformEffect $ logEventWithMultipleParams state.data.logField "ny_user_invoice_download" $ [ { key : "Base fare", value : unsafeToForeign state.data.selectedItem.baseFare},
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/CustomerUtils/InvoiceScreen/Controller.purs:73]: here unsafePerformEffect is not required

**Review by Vignesh-772 — APPROVED** [review]

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/MobilityAppBridge.java [hunk 4]
```diff
@@ -470,6 +470,27 @@ public void setCleverTapUserProp(String key, String value) {
             clevertapDefaultInstance.pushProfile(profileUpdate);
     }
 
+    @JavascriptInterface
+    public void setCleverTapUserMultipleProp(String arr) {
+        if (clevertapDefaultInstance != null) {
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/MobilityAppBridge.java:475]: can you set the clevertap instance if it null

File: Frontend/ui-customer/src/Services/Backend.purs [hunk 5]
```diff
@@ -910,3 +910,12 @@ disabilityList = do
   lift $ lift $ withAPIResult (EP.disabilityList "") unwrapResponse $ callAPI headers GetDisabilityListReq
   where
     unwrapResponse x = x
+------------------------------------------------------------------------ Person Stats ------------------------------------------------------------------------------------
+
+getPersonStatsBT :: String -> FlowBT String PersonStatsRes
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Services/Backend.purs:915]: is it required as it is not used any where

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Review by S-PRAVINKUMAR — CHANGES_REQUESTED** [review]

**Review by Vignesh-772 — APPROVED** [review]

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Vignesh-772 — DISMISSED** [review]

**Review by Rohit4997 — APPROVED** [review]

**Review by S-PRAVINKUMAR — APPROVED** [review]

**Review by Vignesh-772 — APPROVED** [review]

**Pull request merged**